money_honey.py

```diff
@@ -1,80 +1,164 @@
+                                   action=self.currency_check(self.currency_changer),
+                                   help="Three chars code or currency symbol from which we are converting.")
+        args_parser.add_argument("-oc", "--output_currency", required=False,
+                                 action=self.currency_check(self.currency_changer),
+                                 help="Three chars code or currency symbol to which we are converting.")
+        args_parser.add_argument("-nj", "--no_json", required=False, action='store_true',
+                                 help="Will change output format from json to custom one.")
+        self.args = args_parser.parse_args()
+
+    def currency_check(self, currency_changer):
+        """
+        Custom action for argument parser to restrict values of currency inputs.
+        :param currency_changer: Instance of class actual_currencies.CurrenciesChanger.
+        :return: Class which inherits from argparse.Action and takes care of restrictions.
+        """
+        class CurrencyCheck(argparse.Action):
+            def __call__(self, parser, namespace, values, option_string=None):
+                if isinstance(values, str) and len(values) <= 3:
+                    if not currency_changer.is_supported_currency(values.upper()):
+                        parser.error("argument {0}/{1}: '{2}' is not supported currency."
+                                     .format(self.option_strings[0], self.option_strings[1], values))
+                else:
+                    parser.error("argument {0}/{1}: has to be str no longer than 3 chars."
+                                 .format(self.option_strings[0], self.option_strings[1]))
+
+                setattr(namespace, self.dest, values)
+
+        return CurrencyCheck
+
+    def decimal_check(self):
+        """
+        Check for correct decimal. Could have been just class but to keep it consistent I went with func.
+        :return: Decimal value.
+        """
+        class DecimalCheck(argparse.Action):
+            def __call__(self, parser, namespace, values, option_string=None):
+                try:
+                    values = Decimal(values)
+                except InvalidOperation:
+                    parser.error("argument {0}/{1}: has to be Decimal."
+                                 .format(self.option_strings[0], self.option_strings[1]))
+
+                setattr(namespace, self.dest, values)
+
+        return DecimalCheck
+
+    def convert_money(self):
+        """
+        Prepare inputs to get converted followed up with conversion.
+        """
+        self.handle_input_currency()
+        self.handle_output_currency()
+        self.currency_changer.set_exchange_params(self.args.amount, self.args.input_currency, self.args.output_currency)
+        self.converted_amounts = self.currency_changer.get_changed_values()
+
+    def handle_input_currency(self):
+        # if currency is symbol
```

Comment:
This part was before bit messy and Michal said it's [hard to read](https://github.com/ZlatyChlapec/MoneyHoney/pull/1#pullrequestreview-22614245). Since I wasn't able to get rid of many elses I at least separated them to two different functions and commented them so it's easy to read.